src/python/pants/jvm/resolve/coursier_fetch_integration_test.py

```diff
@@ -668,3 +668,60 @@ def test_transitive_excludes(rule_runner: RuleRunner) -> None:
     entries = resolve.entries
     assert any(i for i in entries if i.coord.artifact == "jackson-databind")
     assert not any(i for i in entries if i.coord.artifact == "jackson-core")
+
+
+@pytest.mark.xfail(
+    reason="coursier bug: https://github.com/coursier/coursier/issues/2884", strict=True
+)
+@maybe_skip_jdk_test
+def test_missing_entry_for_transitive_dependency(rule_runner: RuleRunner) -> None:
+    resolve = rule_runner.request(
+        CoursierResolvedLockfile,
+        [
+            ArtifactRequirements(
+                [
+                    Coordinate(
+                        group="org.apache.hive",
+                        artifact="hive-exec",
+                        version="1.1.0",
+                    )
+                    .as_requirement()
+                    .with_extra_excludes(
+                        "org.apache.calcite:calcite-avatica",
+                        "org.apache.calcite:calcite-core",
+                        "jdk.tools:jdk.tools",
+                    )
+                ]
+            )
+        ],
+    )
+
+    lookup = {(entry.coord.group, entry.coord.artifact): entry for entry in resolve.entries}
+    missing = []
+    for entry in resolve.entries:
+        for d in entry.dependencies:
+            coord = (d.group, d.artifact)
+            if coord not in lookup:
+                missing.append(coord)
```

Comment:
This is more stylistic, but it took me a little bit to be sure that this was doing what was expected. Since this is a test and efficiency doesn't matter, one option might be to compute entire explicit sets and diff them directly:

```python
coords_of_entries = {(entry.coord.group, entry.coord.artifact) for entry in resolve.entries}
coords_of_dependencies = {(d.group, d.artifact) for entry in resolve.entries for d in entry.dependencies}
missing = coords_of_dependencies - coords_of_entries

# ... comment ...
assert missing = {("junit", "junit"), ("org.apache.curator", "apache-curator")}
```